src/main/java/org/apache/cassandra/sidecar/models/HttpResponse.java

```diff
@@ -97,17 +97,23 @@ public void setInternalErrorStatus(String msg)
      */
     public Future<Void> sendFile(String fileName, long fileLength, HttpRange range)
     {
-        // notify client we support range requests
-        response.putHeader(HttpHeaderNames.ACCEPT_RANGES, "bytes");
-
-        if (range.length() != fileLength)
-        {
-            setPartialContentStatus(range);
-        }
-
-        return response.putHeader(HttpHeaderNames.CONTENT_TYPE, HttpHeaderValues.APPLICATION_OCTET_STREAM)
-                       .putHeader(HttpHeaderNames.CONTENT_LENGTH, Long.toString(range.length()))
-                       .sendFile(fileName, range.start(), range.length());
+        // Defer setting headers in case of an error before sending the file
+        response.headersEndHandler(v -> {
+            // notify client we support range requests
+            response.putHeader(HttpHeaderNames.ACCEPT_RANGES, "bytes");
+
+            if (range.length() != fileLength)
+            {
+                setPartialContentStatus(range);
+            }
+            response.putHeader(HttpHeaderNames.CONTENT_TYPE, HttpHeaderValues.APPLICATION_OCTET_STREAM)
+                    .putHeader(HttpHeaderNames.CONTENT_LENGTH, Long.toString(range.length()));
```

Comment:
Those 2 headers are put as part of `sendFile`. See `io.vertx.core.http.impl.Http2ServerResponse#sendFile:line#619-627`. And then at line#628 `checkSendHeaders(false);` the `headersEndHandler` is called. 

Other than that, the `CONTENT_LENGTH` might not always be the `range.length()`. It is the minimum of `Math.min(length, file_.length() - offset);` as implemented in `io.vertx.core.http.impl.HttpUtils#resolveFile`

Reply:
ok, maybe we should let vertx set those headers. Thanks for catching it